components/NavigationDrawer/src/private/MDCBottomDrawerContainerViewController.m

```diff
@@ -475,12 +484,34 @@ - (void)viewDidDisappear:(BOOL)animated {
 
 - (void)preferredContentSizeDidChangeForChildContentContainer:(id<UIContentContainer>)container {
   [super preferredContentSizeDidChangeForChildContentContainer:container];
+  if ([container isKindOfClass:[UIViewController class]]) {
+    UIViewController *containerViewController = (UIViewController *)container;
+    if (containerViewController == self.contentViewController) {
+      CGFloat height = containerViewController.preferredContentSize.height;
+      [self resetInitialDrawerFactorWithHeight:height];
+    }
+  }
   _contentHeaderTopInset = NSNotFound;
   _contentHeightSurplus = NSNotFound;
   _addedContentHeight = NSNotFound;
   [self.view setNeedsLayout];
 }
 
+- (void)resetInitialDrawerFactorWithHeight:(CGFloat)height {
+  if (_contentVCPreferredContentSizeHeightCached == 0) {
+    [self cacheLayoutCalculations];
+  }
+  CGFloat totalHeight = self.headerViewController.preferredContentSize.height +
+                        _contentVCPreferredContentSizeHeightCached;
+  CGFloat precentageOfFullScreen = totalHeight / self.presentingViewBounds.size.height;
+  if (precentageOfFullScreen > 0.5) {
+    precentageOfFullScreen = 0.5;
+  }
+  self.initialDrawerFactor = precentageOfFullScreen;
+  CGFloat diff = _contentVCPreferredContentSizeHeightCached - height;
+  [self.scrollView setContentOffset:CGPointMake(0, diff)];
```

Comment:
why do we need to update the scroll view content offset? can you put a comment in the code explaining?